src/Config/Converters/DataSourceConverterFactory.cs

```diff
@@ -103,6 +103,14 @@ public DataSourceConverter(bool replaceEnvVar)
                                         {
                                             optionsSubpropertyValue = null;
                                         }
+                                        else if (reader.TokenType is JsonTokenType.Number)
+                                        {
+                                            optionsSubpropertyValue = reader.GetInt32();
```

Comment:
what happens if this line fails? e.g. when user sets value to 123.45 -> a JSON Number but not an integer.

Reply:
Resolving as no longer valid and we moved to host options.